chains/solana/contracts/programs/ccip-router/src/messages.rs

```diff
@@ -59,10 +59,13 @@ pub struct Any2SVMRampMessage {
     pub header: RampMessageHeader,
     pub sender: Vec<u8>,
     pub data: Vec<u8>,
-    // receiver is used as the target for the two main functionalities
-    // token transfers: recipient of token transfers (associated token addresses are validated against this address)
-    // arbitrary messaging: expected account in the declared arbitrary messaging accounts (2nd in the list of the accounts)
-    pub receiver: Pubkey,
+    // In EVM receiver means the address that all the listed tokens will transfer to and the address of the message execution.
+    // In Solana the receiver is split into two:
+    // Logic Receiver is the Program ID of the user's program that will execute the message
+    pub logic_receiver: Pubkey,
```

Comment:
Wdyt about naming this `message_receiver` instead of `logic_receiver`? It feels closer to CCIP terminology in general, e.g. ccip use cases include "pure messaging" (thus `message_receiver`) vs "token transfer" vs "programmable token transfer" (which is the sum of the previous two)

Alternatively, using more of the Solana terminology, we could name it `cpi_receiver`

Reply:
There is actually no term for CCIP to distinguish the receiver between logic and tokens as for EVM they are both the same.

Naming them `logic_receiver` and `token_receiver` makes it impossible to have ambiguity between the two terms. Someone could argue that any CPI call is a message receiver or cpi receiver, those are more generics names.